dotCMS/src/main/java/com/dotmarketing/business/cache/provider/CacheSizingUtil.java

```diff
@@ -0,0 +1,257 @@
+     * returns the size of an objects class, based on the classes properties
+     * 
+     * @param object
+     * @return
+     */
+    long sizeOfClass(Object object) {
+
+        return ClassLayout.parseClass(object.getClass()).instanceSize();
+    }
+
+    public static boolean useCompressedOops = true;
+
+    public static int retainedSize(Object obj) {
+        return retainedSize(obj, new HashMap<>());
+    }
+
+    @SuppressWarnings("restriction")
+    private static int retainedSize(Object obj, HashMap<Object, Object> calculated) {
+        Object ref = null;
+        try {
+            if (obj == null)
+                throw new NullPointerException();
+            calculated.put(obj, obj);
+            Class<?> cls = obj.getClass();
+            if (cls.isArray()) {
+                int arraysize = UnsafeAccess.UNSAFE.arrayBaseOffset(cls)
+                                + UnsafeAccess.UNSAFE.arrayIndexScale(cls) * Array.getLength(obj);
+                if (!cls.getComponentType().isPrimitive()) {
+                    Object[] arr = (Object[]) obj;
+                    for (Object comp : arr) {
+                        if (comp != null && !isCalculated(calculated, comp))
+                            arraysize += retainedSize(comp, calculated);
+                    }
+                }
+                return arraysize;
+            } else {
+                int objectsize = sizeof(cls);
+                for (Field f : getAllNonStaticFields(obj.getClass())) {
+                    Class<?> fcls = f.getType();
+                    if (fcls.isPrimitive())
+                        continue;
+                    f.setAccessible(true);
+                    ref = f.get(obj);
+                    if (ref != null && !isCalculated(calculated, ref)) {
+                        calculated.put(ref, ref);
+                        int referentSize = retainedSize(ref, calculated);
+                        objectsize += referentSize;
+                    }
+                }
+                return objectsize;
+            }
+        } catch (Throwable e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @SuppressWarnings("restriction")
+    public static int sizeof(Class<?> cls) {
+
+        if (cls == null)
+            throw new NullPointerException();
+
+        if (cls.isArray())
+            throw new IllegalArgumentException();
+
+        if (cls.isPrimitive())
+            return primsize(cls);
+
+        int lastOffset = Integer.MIN_VALUE;
+        Class<?> lastClass = null;
+
+        for (Field f : getAllNonStaticFields(cls)) {
+            if (Modifier.isStatic(f.getModifiers()))
+                continue;
+
+            int offset = (int) UnsafeAccess.UNSAFE.objectFieldOffset(f);
+            if (offset > lastOffset) {
+                lastOffset = offset;
+                lastClass = f.getClass();
+            }
+        }
+        if (lastOffset > 0)
+            return modulo8(lastOffset + primsize(lastClass));
```

Comment:
Codacy found an issue: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6315945)